src/Document/Meta.php

```diff
@@ -21,4 +23,22 @@ public function jsonSerialize()
     {
         return $this->data;
     }
+
+    private function validateObject($object)
+    {
+        foreach ($object as $name => $value) {
+            if (is_string($name) && !$this->isValidMemberName($name)) {
+                throw new \OutOfBoundsException("Not a valid attribute name '$name'");
+            }
+
+            if (is_array($value) || $value instanceof \stdClass) {
+                $this->validateObject($value);
+            }
+        }
+    }
+
+    private function isValidMemberName(string $name): bool
+    {
+        return preg_match('/^(?=[^-_ ])[a-zA-Z0-9\x{0080}-\x{FFFF}-_ ]*(?<=[^-_ ])$/u', $name) === 1;
```

Comment:
Now this line is duplicated here and in `ResourceObject`.I suggest we move this check into a trait unless we can figure out a better solution